megamek/src/megamek/common/Coords.java

```diff
@@ -485,44 +486,51 @@ public boolean isOnHexRow(int direction, @Nullable Coords other) {
     }
 
     /**
-     * Returns a list of all adjacent coordinates (distance = 1), 
+     * Returns a list of all adjacent coordinates (distance = 1),
      * regardless of whether they're on the board or not.
      */
     public ArrayList<Coords> allAdjacent() {
         return (allAtDistance(1));
     }
-    
+
+    /**
+     * Returns a list of all coordinates at the given distance (dist - 1)
+     * and anything less than dist as well.
+     */
+    public ArrayList<Coords> allLessThanDistance(int dist) {
```

Comment:
places in the code that were using the previous version of the function `allAtDistanceOrLess` were changed to use this one instead. This implements the behavior that the previous/bugged version had (same functionality)